lib/puppet/type/rabbitmq_policy.rb

```diff
@@ -72,5 +72,21 @@ def validate_definition(definition)
         raise ArgumentError, "Invalid definition"
       end
     end
+    # If ha-mode is 'exactly', convert ha-params to integer
+    if definition['ha-mode'] == 'exactly'
+      ha_params = definition['ha-params']
+      unless ha_params.to_i.to_s == ha_params
+        raise ArgumentError, "Invalid ha-params '#{ha_params}' for ha-mode 'exactly'"
+      end
+      definition['ha-params'] = ha_params.to_i
```

Comment:
Values should not be modified from the validate methods. Puppet provides a `munge` method for attributes to perform this kind of conversion.